bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/core/QuickTextQuery.java

```diff
@@ -48,7 +48,7 @@ public int getOffset() {
 
 	private boolean caseInsensitive;
 	private String orgPattern; //Original pattern case preserved even if search is case insensitive.
-	private Pattern pattern;
+	private final ThreadLocal<Pattern> patterns;
```

Comment:
Javadoc says: 

> Instances of this class are immutable and are safe for use by multiple concurrent threads

so maybe better compile the pattern only once and pass it as an argument instead of a string?

Reply:
great. now its simpler